content/articles/how-to-perform-containerized-static-application-security-testing(sast)-using-mobsf-within-a-docker-container/index.md

```diff
@@ -0,0 +1,182 @@
+url: /security-testing-using-mobsf-in-docker/
+title: Static Application Security Testing using MobSF and Docker
+description: This article will introduce the reader to Static Application Security Testing using MobSF and Docker.
+author: ruth-mare
+date: 2022-02-22T00:00:00-17:30
+topics: [Security]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/performing-static-application-testing-in-docker/hero.jpg
+    alt: SAST using MobSF and Docker Hero Image
+---
+A vulnerability that is not fixed could lead to dire consequences for the user and owners of an application. SAST is one of the ways used to scan and assess applications for vulnerabilities.
+<!--more-->
+
+### Table of content
+- [Prerequisites](#prerequisites)
+- [What is SAST](#what-is-sast)
+- [Differences between SAST and DAST](#differences-between-sast-and-dast)
+- [Tools used to perform SAST](#tools-used-to-perform-sast)
+- [Step by Step process of SAST](#step-by-step-process-of-sast)
+- [Advantages and Disadvantages of SAST](#advantages-and-disadvantages-of-sast)
+- [Conclusion](#conclusion)
+- [Relevant Material](#relevant-material)
+
+
+### What is SAST
+Static Application Security Testing(SAST) is a method of vulnerability testing that analyses the source code of an application to determine the security vulnerabilities that may cause the application to be compromised.
+
+> A vulnerability is any loophole within an application that can be exploited to gain unauthorized access to the application or propagate malware through the application. 
+A vulnerability that is not fixed could lead to dire consequences for the user and owners of an application. SAST is one of the ways used to scan and assess applications for vulnerabilities.
+
+SAST scans applications before the compilation of the code right at the beginning of the SDLC because it does not require an executing application.
+
+The above scenario is also known as white-box testing.
+
+Developers can therefore identify shortcomings of an application in the early stages of development and quickly resolve issues that may otherwise affect the final rolled out application.
+
+Developers can systematically track the vulnerability trends within the application thereby promoting the creation of a secure and effecint Software Development Life Cycle(SDLC).
+
+SAST provides visual representations of the problems identified making it easy to navigate the code. In addition, it provides suggestions on how to fix the vulnerabilities identified within the code. 
+
+Customized reports downloadable and visible offline are supported by most SAST tools and can also be tracked through dashboards.
+
+### Differences between SAST and DAST
+There are two different types of Application Security Testing, i.e., SAST and DAST (Dynamic Application Security Testing).
+
+The above-mentioned detect security vulnerabilities in applications except in a unique way for each. Below are some of the critical comparisons of SAST between DAST:
```

Comment:
Consider simplifying this line to better convey the contrast between SAST and DAST